packages/in_app_purchase/in_app_purchase_storekit/lib/src/store_kit_wrappers/sk_product_wrapper.dart

```diff
@@ -228,12 +253,14 @@ class SKProductDiscountWrapper {
         other.priceLocale == priceLocale &&
         other.numberOfPeriods == numberOfPeriods &&
         other.paymentMode == paymentMode &&
-        other.subscriptionPeriod == subscriptionPeriod;
+        other.subscriptionPeriod == subscriptionPeriod &&
+        other.identifier == identifier &&
+        other.type == type;
   }
 
   @override
-  int get hashCode => hashValues(
-      price, priceLocale, numberOfPeriods, paymentMode, subscriptionPeriod);
+  int get hashCode => hashValues(price, priceLocale, numberOfPeriods,
+      paymentMode, subscriptionPeriod, identifier, type);
```

Comment:
Are the equality and hash changes here being tested somewhere?

Reply:
Yes, it has been tested.

Changes of the test stub code:
packages/in_app_purchase/in_app_purchase_storekit/test/store_kit_wrappers/sk_test_stub_objects.dart

Related unit test in line 104 of:
packages/in_app_purchase/in_app_purchase_storekit/test/store_kit_wrappers/sk_product_test.dart